mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2354,7 +2355,7 @@ pub async fn check_balance_for_taker_swap(
     let params = match prepared_params {
         Some(params) => params,
         None => {
-            let dex_fee = dex_fee_amount_from_taker_coin(my_coin, other_coin.ticker(), &volume);
+            let dex_fee = dex_fee_amount_from_taker_coin(my_coin, other_coin.ticker(), &volume).fee_amount();
             let fee_to_send_dex_fee = my_coin
                 .get_fee_to_send_taker_fee(dex_fee.to_decimal(), stage.clone())
                 .await
```

Comment:
As mentioned here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2006#discussion_r1397397808, in `get_fee_to_send_taker_fee`, the `dex_fee_amount` parameter should be `DexFee` not `BigDecimal` to handle the below case correctly https://github.com/KomodoPlatform/komodo-defi-framework/blob/38ae93c6143e19fa58a28963eb025678aa9b3580/mm2src/coins/utxo/utxo_common.rs#L4080-L4087

Reply:
Updated as https://github.com/KomodoPlatform/komodo-defi-framework/blob/2f553d1d2283f30d5fcbde1c506e89c643ca2938/mm2src/coins/utxo/utxo_common.rs#L4054-L4075

with https://github.com/KomodoPlatform/komodo-defi-framework/commit/2f553d1d2283f30d5fcbde1c506e89c643ca2938#diff-51555c8bf8371dcceec68515eaa0a1e44949fff221c965cbbdb7b97de463cb4c

Reply:
Thanks for the update! `get_fee_to_send_taker_fee` LGTM now!